apps/04TowersOfHanoi.js

```diff
@@ -19,24 +19,51 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  // This function works like copy and paste does on a computer. It pops off a block from the start stack and saves it to a variable. then we push that variable to endStack.
+  var start = stacks[startStack].pop();
+  stacks[endStack].push(start);
 }
 
-function isLegal() {
-  // Your code here
+function isLegal(startStack, endStack) {
+  // Both startBlock and endBlock are defined to determine the index of a specific block within the nested arrays by finding the length of the specific stack within the stacks array and subtracting that number by 1 to find the last item in that particular stack's array.
+  // I used only if statements below instead of else if statements, as else if statements tend to depend on each other.
+  // I defined legal as false and made the function return legal at the end just in case something went wrong within the function. This would not allow the player to move forward. Within my if statements, legal changes to true if all of the correct conditions are met.
+  var legal = false
+  var startBlock = stacks[startStack][stacks[startStack].length - 1]
+  var endBlock = stacks[endStack][stacks[endStack].length - 1]
 
-}
+  if (stacks[endStack].length >= 0) {
+    legal = true;
+  }
 
-function checkForWin() {
-  // Your code here
+  if (startBlock < endBlock) {
+    legal = true;
+  }
 
+  if (startBlock > endBlock) {
+    legal = false;
+  }
+
+  return legal;
 }
 
-function towersOfHanoi(startStack, endStack) {
-  // Your code here
+function checkForWin(endStack) {
+  //I originally wanted to say if (stacks[endStack].length) === 4 {} as it's much more clean that what I wrote below, however because of the way the tests are written, this does not work. The test define's the stack instead of playing the game all the way through. Instead I check if each either stack 'b' or 'c' have 4 blocks, then the player has won.
+  if (stacks['b'].length === 4 || stacks['c'].length === 4) {
+    console.log("You Won!!!")
+    return true;
+  } else {
+    return false;
+  }
+  p
+}
 
+function towersOfHanoi(startStack, endStack) {
+  // Each turn this function will first check if that move is legal. If true, then it will move the piece. After moving the piece, it will check if the player has won. If the play wins, it will announce "You Won!!!" If the player has not won, the game will continue.
```

Comment:
I really like this comment, but we didn't do it in the function! Esp all that stuff about
> if true, then it will move the piece